### PR TITLE
ALBS-74: Some improvements for the mirrors service

### PR DESCRIPTION
**Pull request by soksanichenko**
- [Improvement] The mirrors from request's country should be first in the list
- [Improvement] Downgrade the level of logging of issue "Error retrieving Nominatim data for..." to warning
- Refactoring
  - Make context manager for Redis async engine
  - Some logged events should be logged as exceptions not errors
  - Typing according to version of Python (3.9)
  - Docstrings

### CODE REVIEW

File: src/backend/api/utils.py [hunk 1]
```diff
@@ -185,10 +187,9 @@ async def get_azure_subnets_json(http_session: ClientSession) -> Optional[Dict]:
             response_text = await resp.text()
     except client_exceptions.ClientConnectorError as err:
         logger.error(
```
Comment (anfimovdm) [inline src/backend/api/utils.py:189]: May using `exception` method be better than `error` when catching exception?

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: src/backend/api/utils.py (hunk 1 above)
Comment (soksanichenko) [reply src/backend/api/utils.py]: Done

**Review by anfimovdm — APPROVED** [review]

**Review by Korulag — APPROVED** [review]

**Review by anfimovdm — APPROVED** [review]

File: ci/ansible/roles/deploy/defaults/main/packages.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - geoip2==4.4.0
+  - haversine==2.5.1
+  - jsonschema==3.2.0
+  - Jinja2==3.0.2
+  - lxml==4.6.3
+  - pyyaml==6.0
+  - requests==2.26.0
+  - sentry-sdk[flask]==1.4.3
+  - sqlalchemy==1.4.26
+  - uwsgi==2.0.20
```
Comment (Korulag) [inline ci/ansible/roles/deploy/defaults/main/packages.yml:36]: Mirror service still using uwsgi?

File: ci/ansible/roles/deploy/defaults/main/packages.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  - Flask-API==3.0.post1
+  - geoip2==4.4.0
+  - haversine==2.5.1
+  - jsonschema==3.2.0
+  - Jinja2==3.0.2
+  - lxml==4.6.3
+  - pyyaml==6.0
+  - requests==2.26.0
+  - sentry-sdk[flask]==1.4.3
+  - sqlalchemy==1.4.26
```
Comment (Korulag) [inline ci/ansible/roles/deploy/defaults/main/packages.yml:35]: 1.4.27 already available

Comment (soksanichenko) [reply ci/ansible/roles/deploy/defaults/main/packages.yml]: Done

File: ci/ansible/roles/deploy/defaults/main/packages.yml (hunk 2 above)
Comment (soksanichenko) [reply ci/ansible/roles/deploy/defaults/main/packages.yml:36]: Yep. I'll replace it by gunicorn when I migrate from SQLite to PosgreSQL

**Pull request merged**